docs/10-WebServices.md

```diff
@@ -41,20 +63,38 @@ $I = new ApiTester($scenario);
 $I->wantTo('create a user via API');
 $I->amHttpAuthenticated('service_user', '123456');
 $I->haveHttpHeader('Content-Type', 'application/x-www-form-urlencoded');
-$I->sendPOST('users', ['name' => 'davert', 'email' => 'davert@codeception.com']);
+$I->sendPOST('/users', ['name' => 'davert', 'email' => 'davert@codeception.com']);
 $I->seeResponseCodeIs(200);
 $I->seeResponseIsJson();
 $I->seeResponseContains('{"result":"ok"}');
 ?>
 ```
 
-REST module is designed to be used with services that serve responses in JSON format. For example, method `seeResponseContainsJson` will convert provided array to JSON and check whether response contains it.
+### Testing JSON Responses
+
+The latest line checked that the response contained provided string. However we shoudn't rely on it, as dpending on content formatting we can receive different results with the same data. What we actually need it to check that response can be parsed and it contains some of values we expect. In case of JSON we can use `seeResponseContainsJson` method
 
-You may want to perform more complex assertions on response. This can be done with writing your own methods in [Helper](http://codeception.com/docs/03-ModulesAndHelpers#Helpers) classes. To access the latest JSON response you will need to get `response` property of `REST` module. Let's demonstrate it with `seeResponseIsHtml` method:
+``` php
+<?php
+// matches {"result":"ok"}'
+$I->seeResponseContainsJson(['result' => 'ok']);
+// it can match tree-like strcutcures as well
+$I->seeResponseContainsJson([
+	'user' => [
+			'name' => 'davert',
+			'email' => 'davert@codeception.com',
+			'status' => 'inactive'
+	]
+]);
+?>
+```
+
+You may want to perform even more complex assertions on response. This can be done with writing your own methods in [Helper](http://codeception.com/docs/03-ModulesAndHelpers#Helpers) classes. To access the latest JSON response you will need to get `response` property of `REST` module. Let's demonstrate it with `seeResponseIsHtml` method:
```

Comment:
complex assertions on **a** response

be done **~~with~~by** writing your

response you will need to get **the** `response` property of **the** `REST` module